src/lib/FreeC/LiftedIR/Effect.hs

```diff
@@ -0,0 +1,9 @@
+-- | This module contains a data type for effect constraints.
+
+module FreeC.LiftedIR.Effect where
+
+-- | An effect constraint.
+--
+--   These effects corresponds to type classes constraining the container used by
+--   the @Free@ monad. Currently just partiality is supported.
+data Effect = Partiality deriving (Show, Eq)
```

Comment:
I think we should try to link the effect constructors of the lifted IR to the effect defined here. Otherwise there is no guarantee that a program contains only the effects listed in its effect set.

Reply:
Although a guarantee like this would be nice to have, it is probably too involved to implement, especially considering that it is basically an elaborate test of the effect set annotation algorithm.